### DISCUSSION TITLE
Sidecar container for kafka and zookeeper

### DISCUSSION THREAD

**Discussion by prateekkohli21** [strimzi/strimzi-kafka-operator#7499] [Q&A]
Hi,

I would like to add FluentBit(https://github.com/leahnp/fluentbit-sidecar) as a sidecar to collect logs for kafka and zookeeper.
But I guess Strimzi does not allow any such provision to add a sidecar container.

Is there any possible way to support it?

**Comment by scholzj** [comment]
No, you cannot add sidecars to the Pods. It is not really clear to me why would it be useful -> the logging is anyway done to standard output and you can capture it with Fluentbit from the container engine logs.

**Reply by prateekkohli21** [reply]
Thanks for your response!!

Actually we had 2 options:
1) To deploy fluentbit as daemon set
2) To deploy fluentbit as a sidecar

But we do not want to use the daemon set option as we will have to provide root privileges to fluent bit daemon set container, which will not be allowed in our case.

So because of this reason we wanted to deploy fluent bit as a sidecar.

Would it be possible for you to suggest the area in the Strimzi code from where we can start looking if we want to modify it to add a new sidecar?

**Reply by scholzj** [reply]
I'm not sure there is a single area. This is not completely trivial as it needs multiple things which touch different parts. You need the sidecar, you need the storage shared between them, you need to handle rolling updates and readiness, you need to have the logging actually log into a file etc. Sticking with the _usual_ ways to do things such as DaemonSet is better and would save you these troubles in the long term (such as rewriting the above for new releases to upgrade etc.).
